src/Npgsql/NpgsqlDataSourceConfiguration.cs

```diff
@@ -10,6 +9,7 @@ namespace Npgsql;
 sealed record NpgsqlDataSourceConfiguration(string? Name,
     NpgsqlLoggingConfiguration LoggingConfiguration,
     NpgsqlTracingOptions? TracingOptions,
+    NpgsqlTypeLoadingOptions TypeLoading,
     TransportSecurityHandler TransportSecurityHandler,
     IntegratedSecurityHandler userCertificateValidationCallback,
```

Comment:
While you're changing things around, could you please rename `userCertificateValidationCallback` to `IntegratedSecurityHandler`?